web/src/components/profile/getErrorStateForAddressField.ts

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import {
+  BUSINESS_ADDRESS_LINE_1_MAX_CHAR,
+  BUSINESS_ADDRESS_LINE_2_MAX_CHAR,
+} from "@/components/tasks/business-formation/getErrorStateForFormationField";
+import { getMergedConfig } from "@/contexts/configContext";
+import { isZipCodeNj } from "@/lib/domain-logic/isZipCodeNj";
+import { AddressFieldErrorState, AddressFields, FieldsForAddressErrorHandling } from "@/lib/types/types";
+import { templateEval } from "@/lib/utils/helpers";
+import { Address } from "@businessnjgovnavigator/shared/index";
+
+export const getErrorStateForAddressField = ({
```

Comment:
[Future Refactor] Given the commonalities between this use case, formation, and the address modal... some of the functions here could get carved into an address helpers file.

Reply:
created ticket